### PR TITLE
[DOCS] Expand examples for runtime fields in a search query

### PR DESCRIPTION
**Pull request by lockewritesdocs**
This PR adds more examples for adding runtime fields in a search query, and shows to create runtime fields that reference other runtime fields.

Preview link: https://elasticsearch_71237.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/runtime-search-request.html#runtime-search-request-examples

### CODE REVIEW

File: docs/reference/mapping/runtime.asciidoc [hunk 1]
```diff
@@ -199,18 +199,18 @@ to `boolean`, or if the runtime field is removed.
...
-the same name in the index mappings. This flexibility allows you to shadow
-existing fields and calculate a different value in the search request, without
-modifying the field itself. If you made a mistake in your index mapping, you
-can use runtime fields to calculate values that override values in the mapping
-during the search request.
+Defining a runtime field in a search request uses the same format as defining
+a runtime field in the index mapping. That consistency means you can promote a
+runtime field from a search request to the index mapping by moving the field
+definition from `runtime_mappings` in the search request to the `runtime`
+section of the index mapping.
```
Comment (nik9000) [inline docs/reference/mapping/runtime.asciidoc:209]: I think it'd be less "product-y" to say something like "Defining a runtime field in a search request uses the same format as defining a runtime field in the index mapping. So you can copy and paste from one to the other."  Or something like that. I dunno. That's partly me putting my casual voice into it. Maybe its just me, but I think its kind of obvious that we did this intentionally and we don't need to sounds pedantic about it? I dunno.

File: docs/reference/mapping/runtime.asciidoc [hunk 2]
```diff
@@ -235,11 +235,161 @@ GET my-index/_search
...
+      "value" : 333.3333333333333
+    },
+    "avg_end" : {
+      "value" : 8658642.333333334
+    }
+  }
+}
+----
+// TESTRESPONSE[s/\.\.\./"took" : $body.took,"timed_out" : $body.timed_out,"_shards" : $body._shards,/]
+// TESTRESPONSE[s/\+\+\+/"hits" : $body.hits,/]
```
Comment (nik9000) [inline docs/reference/mapping/runtime.asciidoc:333]: Maybe use `filter_path` to make this a little easier. Like this: https://github.com/elastic/elasticsearch/pull/71202/files#diff-b6773f30037d9ee16418b7d8645ccd85f425a16ea7caa5d6c49a55060b62a885R198

File: docs/reference/mapping/runtime.asciidoc (hunk 1 above)
Comment (lockewritesdocs) [reply docs/reference/mapping/runtime.asciidoc:209]: That's a good callout. How about something like:
>Defining a runtime field in a search request uses the same format as defining a runtime field in the index mapping.
Just copy the field definition from the `runtime_mappings` in the search request to the `runtime` section of the index mapping.

Comment (nik9000) [reply docs/reference/mapping/runtime.asciidoc:209]: Sounds good to me!

**Review by nik9000 — APPROVED** [review]

**Pull request merged**